data_structures/queue_using_array.cpp

```diff
@@ -1,69 +1,69 @@
 /*
     Write a program to implement Linear Queue using array.
-
     Functions to implement
         Enqueue (Insertion)
         Dequeue (Deletion)
-
 */
 #include <iostream>
+#include <vector>
 
-#define MAXSIZE 10
+constexpr int max_size = 10;
 
 class Queue_Array {
-    int front;
-    int rear;
-    int size;
-
- public:
-    Queue_Array() {
-        front = -1;
-        rear = -1;
-        size = MAXSIZE;
-    }
-    int *arr = new int[size];
+public:
+    Queue_Array();
     void enqueue(int);
     int dequeue();
-    void display();
+    void display() const;
+private:
+    int front;
+    int rear;
+    std::vector<int> arr;
```

Comment:
```suggestion
    int front = 0;
    int rear = 0;
    std::vector<int> arr{};
```

Reply:
Dear @Panquesito7 ,
I changed them as you said in upcoming commit, but I have a question for my curiosity:
Should we initialize private member variables when we declare them or in class's constructor? Can I know why for further information? :D
Thanks for your help

Reply:
> when we declare them or in class's constructor?

I think it's better to initialize them first when you declare them, and you can
then re-initialize them in the class constructor. It should not harm the code.